contrib/generate_precompile.jl

```diff
@@ -141,7 +141,10 @@ if Artifacts !== nothing
     precompile_script *= """
     using Artifacts, Base.BinaryPlatforms, Libdl
     artifacts_toml = abspath(joinpath(Sys.STDLIB, "Artifacts", "test", "Artifacts.toml"))
-    # cd(() -> (name = "HelloWorldC"; @artifact_str(name)), dirname(artifacts_toml))
```

Comment:
One reason this didn't work before is that the macro is expanded before the `cd` happens. But even so, it won't find `HelloWorldC` as an artifact. So I changed to `macroexpand`.